examples/notebooks/beam-ml/per_key_models.ipynb

```diff
@@ -0,0 +1,597 @@
+        "    (\"I feel fine\", \"neutral\"),\n",
+        "    (\"I love chocolate\", \"positive\"),\n",
+        "]"
+      ],
+      "metadata": {
+        "id": "5HAziWEavQws"
+      },
+      "execution_count": 5,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "To feed our examples into RunInference, we need to have distinct keys that can easily map to our model. In this case, we will define keys of the form `<model_name>-<actual_sentiment>` so that we can extract the actual sentiment of the example later."
+      ],
+      "metadata": {
+        "id": "r6GXL5PLFBY7"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "class FormatExamples(beam.DoFn):\n",
+        "  \"\"\"\n",
+        "  Map each example to a tuple of ('<model_name>-<actual_sentiment>', 'example').\n",
+        "  We will use these keyes to map our elements to the correct models.\n",
+        "  \"\"\"\n",
+        "  def process(self, element: Tuple[str, str]) -> Iterable[Tuple[str, str]]:\n",
+        "    yield (f'distilbert-{element[1]}', element[0])\n",
+        "    yield (f'roberta-{element[1]}', element[0])"
+      ],
+      "metadata": {
+        "id": "p2uVwws8zRpg"
+      },
+      "execution_count": 6,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Using the formatted keys, we will define a `KeyedModelHandler` which maps keys to the ModelHandler we should use for those keys. `KeyedModelHandler` also allows you to define an optional `max_models_per_worker_hint` which will limit the number of models that can be held in a single worker process at once. This is useful if you are worried about your worker running out of memory. See https://beam.apache.org/documentation/sdks/python-machine-learning/index.html#use-a-keyed-modelhandler for more info on managing memory."
+      ],
+      "metadata": {
+        "id": "IP65_5nNGIb8"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "per_key_mhs = [\n",
+        "    KeyModelMapping(['distilbert-positive', 'distilbert-neutral', 'distilbert-negative'], distilbert_mh),\n",
+        "    KeyModelMapping(['roberta-positive', 'roberta-neutral', 'roberta-negative'], roberta_mh)\n",
+        "]\n",
+        "mh = KeyedModelHandler(per_key_mhs, max_models_per_worker_hint=2)"
+      ],
+      "metadata": {
+        "id": "DZpfjeGL2hMG"
+      },
+      "execution_count": 7,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Postprocess our results\n",
+        "\n",
+        "The `RunInference` transform returns a Tuple of the original key and a `PredictionResult` object that contains the original example and the inference. From that, we will extract the data we care about. We will then group this data by the original example in order to compare each model's prediction."
```

Comment:
```suggestion
        "The `RunInference` transform returns the following items:\n",
        "\n",
        "* A tuple of the original key\n",
        "* A `PredictionResult` object that contains the original example and the inference\n",
        "\n",
        "Use those outputs to extract the relevant data. Then, to compare each model's prediction, group this data by the original example."
```

Reply:
That changes the meaning (which means it definitely wasn't clear 😅 ). Let me know if it reads better now

Reply:
This is the relevant diff - https://github.com/apache/beam/pull/28327/commits/056d6d2ebc509938686497e1a0baf8946cdcf136